vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthSplashCarouselFragment.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.onboarding.ftueauth
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import com.airbnb.mvrx.withState
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.google.android.material.tabs.TabLayoutMediator
+import dagger.hilt.android.AndroidEntryPoint
+import im.vector.app.BuildConfig
+import im.vector.app.R
+import im.vector.app.core.resources.StringProvider
+import im.vector.app.databinding.FragmentFtueSplashCarouselBinding
+import im.vector.app.features.VectorFeatures
+import im.vector.app.features.onboarding.OnboardingAction
+import im.vector.app.features.onboarding.OnboardingFlow
+import im.vector.app.features.settings.VectorPreferences
+import org.matrix.android.sdk.api.failure.Failure
+import java.net.UnknownHostException
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class FtueAuthSplashCarouselFragment : AbstractFtueAuthFragment<FragmentFtueSplashCarouselBinding>() {
+
+    @Inject lateinit var vectorPreferences: VectorPreferences
+    @Inject lateinit var vectorFeatures: VectorFeatures
+    @Inject lateinit var carouselController: SplashCarouselController
+    @Inject lateinit var stringProvider: StringProvider
```

Comment:
Why not injected in the constructor?

Reply:
(Also a stringProvider here is a bit useless)

Reply:
wasn't sure if we wanted to use it for consistency (can be helpful if the string provider ends up with extra behaviour)

didn't realise we had setup fragment constructor injection, will try it

Reply:
1- You are directly using `getString()` later in the file :)
2- Cool thanks

Reply:
done and done! https://github.com/vector-im/element-android/pull/4727/commits/311474db1a8e1a3d6fb11f1e8c2c2a93dba78f49 https://github.com/vector-im/element-android/pull/4727/commits/cb88bfd7161fde08e45c327a3defb97d46239b91